src/qt/peerdialog.cpp

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2011-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "guiutil.h"
+#if defined(HAVE_CONFIG_H)
+#include "config/bitcoin-config.h"
+#endif
+
+#include <iostream>
+#include <string>
+
+#include "peerdialog.h"
+
+#include "ui_addpeerdialog.h"
+#include "ui_testpeerdialog.h"
+
+#include "net.h"
+#include "net_processing.h"
+#include "netbase.h"
+#include "protocol.h"
+#include "chainparams.h"
+#include "util.h"
+
+#include <stdio.h>
+
+#include <QMessageBox>
+#include <QHostAddress>
+#include <QAbstractSocket>
+#include <QUrl>
+
+/** Function to manage peers */
+QString PeerTools::ManagePeer(QString type, QString peer)
+{
+    std::string peerAddress = peer.toStdString();
+
+    if(!g_connman)
+        return "Error: Peer-to-peer functionality missing or disabled";
+
+    if (type == "onetry")
+    {
+        CAddress addr;
+        g_connman->OpenNetworkConnection(addr, false, NULL, peerAddress.c_str());
+        return "Attempted to one try node.";
+    }
+
+    if (type == "add")
+    {
+        if(!g_connman->AddNode(peerAddress))
+            return "Error: Node already added";
+    }
+    else if(type == "remove")
+    {
+        if(!g_connman->RemoveAddedNode(peerAddress))
+        {
+            if(!g_connman->DisconnectNode(peerAddress))
+                return "Node not found in connected nodes";
+
+            return "Disconnected the node: " + peer;
+        }
+        else
+        {
+            if(!g_connman->DisconnectNode(peerAddress))
+                return "Node not found in connected nodes";
+        }
+    }
+
+    return "Returned OK.";
+}
+
+/** Check if Peer is vaild */
+bool PeerTools::CheckPeerAddress(QString address)
+{
+    if(PeerTools::CheckIPAddress(address))
```

Comment:
I haven't tested this to be sure, but would `LookupHost()` in netbase.cpp be a good way to validate the address perhaps? It looks like it basically is how the node will resolve it anyway, so would be identicalish validation to what happens later.